optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolver.java

```diff
@@ -135,12 +140,33 @@ public boolean isTerminateEarly() {
 
     @Override
     public boolean addProblemFactChange(ProblemFactChange<Solution_> problemFactChange) {
-        return basicPlumbingTermination.addProblemFactChange(problemFactChange);
+        return basicPlumbingTermination.addProblemChange(new ProblemFactChangeAdapterImpl<>(problemFactChange));
     }
 
     @Override
     public boolean addProblemFactChanges(List<ProblemFactChange<Solution_>> problemFactChangeList) {
-        return basicPlumbingTermination.addProblemFactChanges(problemFactChangeList);
+        Objects.requireNonNull(problemFactChangeList, "The list of problem fact changes ("
+                + problemFactChangeList + ") cannot be null.");
+        List<ProblemChangeAdapter<Solution_>> adaptedProblemChangeList = problemFactChangeList.stream()
+                .map(ProblemFactChangeAdapterImpl::new)
+                .collect(Collectors.toList());
+        return basicPlumbingTermination.addProblemChanges(adaptedProblemChangeList);
```

Comment:
ProblemFactChange is now old, ProblemChange is new.
Philosophy: don't covert-wrap to new to the old, so the old impl just keeps working.
Instead, convert-wrap the old into the new, and adjust the old impl code to work a new only, so the new is as fast as possible.

Not sure if that's possible here (PC might not be able to do everything that PFC does), in which case ignore this comment.

Reply:
Ok, nevermind this - neither of those 2 cases applies, there a 3th form (ProblemFactChangeAdapter) to deal with the fact that a PFC is richer than a PC. Looks good as is!